Makefile

```diff
@@ -546,10 +546,19 @@ generate-examples-clusterclass: $(KUSTOMIZE) clean-examples ## Generate examples
 ## Docker
 ## --------------------------------------
 ##@ docker buils:
-
 .PHONY: docker-build
 docker-build: ## Build the docker image for controller-manager
-	docker build --network=host --pull --build-arg ARCH=$(ARCH) . -t $(CONTROLLER_IMG)-$(ARCH):$(TAG)
+	docker build --network=host --pull \
+	--build-arg LDFLAGS="-s -w -extldflags=-static" \
+	--build-arg ARCH=$(ARCH) . -t $(CONTROLLER_IMG)-$(ARCH):$(TAG)
+	MANIFEST_IMG=$(CONTROLLER_IMG)-$(ARCH) MANIFEST_TAG=$(TAG) $(MAKE) set-manifest-image
+	$(MAKE) set-manifest-pull-policy
+
+.PHONY: docker-build-debug
+docker-build-debug: ## Build the docker image for controller-manager with debug info
+	docker build --network=host --pull \
+	--build-arg LDFLAGS="-extldflags=-static" \
+	--build-arg ARCH=$(ARCH) . -t $(CONTROLLER_IMG)-$(ARCH):$(TAG)
```

Comment:
(Not related to the change in PR) The tag were producing here is funny, we never publish the arch and the tag is our API version, which we never use anywhere either.